### PR TITLE
mpu9250: fix sometimes mpu9250 fails to initialize.

### PR DESCRIPTION
**Pull request by iZhangHui**
@julianoes Please review this PR.
It should fix issue #89 .  

Issue reproduce
Run a simple script like below:
```
#!/bin/bash
for i in {1..1000}
do
./df_imu_test
done

```
Issue arise by luck. :)

When issue happened, run df_imu_test with my fix, then MPU9250 will work fine.

Attachment is my test log.
screenshot from 2016-07-04 13-40-56

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Awesome, thanks!

File: drivers/mpu9250/MPU9250.hpp [hunk 1]
```diff
@@ -250,6 +250,11 @@ class MPU9250: public ImuSensor
 		return _readReg(address, val);
 	}
 
+	int modifyReg(uint8_t address, uint8_t clearbits, uint8_t setbits)
```
Comment (julianoes) [inline drivers/mpu9250/MPU9250.hpp]: I would argue this wrapper function is not used/needed, right?

Comment (julianoes) [inline drivers/mpu9250/MPU9250.hpp]: Nevermind, found it.

File: framework/src/SPIDevObj.cpp [hunk 2]
```diff
@@ -252,6 +252,23 @@ int SPIDevObj::_writeReg(uint8_t address, uint8_t val)
 #endif
 }
 
+int SPIDevObj::_modifyReg(uint8_t address, uint8_t clearbits, uint8_t setbits)
```
Comment (julianoes) [inline framework/src/SPIDevObj.cpp]: Cool, thanks for that!

File: drivers/mpu9250/MPU9250.cpp [hunk 3]
```diff
@@ -343,20 +344,25 @@ void MPU9250::reset_fifo()
...
+		DF_LOG_ERR("FIFO reset failed");
 	}
+}
+
+void MPU9250::clear_int_status()
+{
+	int result;
+	uint8_t int_status = 0;
+
+	result = _readReg(MPUREG_INT_STATUS, int_status);
```
Comment (julianoes) [inline drivers/mpu9250/MPU9250.cpp]: I guess this is all it needed, awesome! Thanks a lot, I was debugging this for several days and couldn't figure it out.

**Pull request merged**